packages/fxa-settings/src/components/ConnectedServices/index.tsx

```diff
@@ -112,31 +102,26 @@ export const ConnectedServices = () => {
     hideAdviceModal,
   ] = useBooleanState();
 
+  // You'd think that you could have client be from useState but due to how useMutation works
+  // the value of the state inside the update function is unpredicatable. A ref is more
+  // manual but totally predictable.
+  const client = useRef<AttachedClient>();
+  const [reason, setReason] = useState<string>();
+
   const clearDisconnectingState = (
     errorMessage?: string,
     error?: ApolloError
   ) => {
     hideConfirmDisconnectModal();
-    DS.client = null;
-    DS.reason = null;
+    client.current = undefined;
+    setReason(undefined);
```

Comment:
Optional: it can be nice to set things to `null` rather than `undefined`, so that when debugging, it's easy to tell if a variable was accidentally never set (undefined), or was deliberately set to a falsy value (null).